### PR TITLE
Run go test on CI

### PR DESCRIPTION
**Pull request by tomleb**
# Issue https://github.com/rancher/rancher/issues/44445

We do have a few tests in apiserver and those are not being run in the CI.

This PR adds 2 drone pipeline, one for amd64 and one for arm64, which simply runs `go test ./...`.

The next thing would be to add golangci-lint. golangci-lint currently fails for a few reasons: ignored errors, unused variables, variables overwritten, etc. I'm not knowledgeable enough to fix those while being confident that I don't introduce regressions.

### CODE REVIEW

File: scripts/test [hunk 1]
```diff
@@ -0,0 +1,7 @@
+#!/bin/bash
+set -e
+
+cd $(dirname $0)/..
+
+echo Running tests
+go test ./...
```
Comment (maxsokolovsky) [inline scripts/test:7]:
```suggestion
go test ./... -race -cover
```

**Review by maxsokolovsky — COMMENTED** [review]
Is this all temporary, since we might move such pipelines to GitHub Actions soon?

**Comment by tomleb** [comment]
> Is this all temporary, since we might move such pipelines to GitHub Actions soon?

Yeah, I'd still rather have tests run in CI though since I don't know what's the timeline for github actions. WDYT?

File: scripts/test (hunk 1 above)
Comment (tomleb) [reply scripts/test]: Added as suggested

**Review by maxsokolovsky — APPROVED** [review]
Looks good! Would we need the same on other branches?

**Comment by tomleb** [comment]
> Looks good! Would we need the same on other branches?

I'm planning to backport to 2.7 and 2.8 at least, not sure about 2.6.

**Review by maxsokolovsky — APPROVED** [review]

**Comment by tomleb** [comment]
> Is this all temporary, since we might move such pipelines to GitHub Actions soon?

Looks like this is needed sooner than expected so I'll look to see how this can be migrated to GH action instead.

**Comment by tomleb** [comment]
Discussed with @MbolotSuse and we also rather not introduce dapper to new repos that don't have it. I'll be closing this PR and open another one which doesn't include dapper but does include a Makefile to try to have "standard" commands (eg: test).

**Pull request closed without merging**